WebSite.json

```diff
@@ -92,7 +92,8 @@
                     "name": "appsettings",
                     "type": "config",
                     "dependsOn": [
-                        "[resourceId('Microsoft.Web/Sites', parameters('webSiteName'))]"
+                        "[resourceId('Microsoft.Web/Sites', parameters('webSiteName'))]",
+						 "Microsoft.ApplicationInsights.AzureWebSites"

```

Comment:
Are you sure this dependency is specified correctly? I'm somewhat surprised it works.....

Reply:
To be honest with you I was quite surprised myself as well, but I tried it multiple times and the deployment always succeeded this way.